go/autoscaler/autoscaler.go

```diff
@@ -282,6 +309,11 @@ func scaleDryRun(r *ClusterResource, j job, curDiff int, maxLoadDesired float64,
 		return
 	}
 
+	if !searchAssignableNodeByCPU(r, j) || !searchAssignableNodeByMem(r, j) {
+		// can not find assignable node, do not scale
+		additional = 0
```

Comment:
Need to return immediately from here, or else the code below may be executed.

Reply:
Done.